clients/src/main/java/org/apache/kafka/common/security/plain/PlainSaslServer.java

```diff
@@ -155,7 +149,7 @@ public SaslServer createSaslServer(String mechanism, String protocol, String ser
             if (!PLAIN_MECHANISM.equals(mechanism)) {
                 throw new SaslException(String.format("Mechanism \'%s\' is not supported. Only PLAIN is supported.", mechanism));
             }
-            return new PlainSaslServer(cbh);
+            return new PlainSaslServer(cbh, new JaasPlainPasswordVerifier());
```

Comment:
I can write a patch to make it pluggable from a config if there's enough interest to review and merge such patch. But this change alone itself allows people to instantiate `PlainSaslServer` with their own implementation of `SaslPasswordVerifier` without modifying `PlainSaslServer` implementation and without any need of new configs. @edoardocomar